ingress: Correct FromGroups rule Parsing

Alex-Waring opened the pull request:
<!-- Description of change -->

In https://github.com/cilium/cilium/pull/30708 I created a resource that translates to a K8s resource but not to an actual resource. This PR corrects that behaviour by adding in some tests and fixing when they break. You could argue I should have done this initially, you're right.

This should be backported to 1.16 so it goes out with the original PR.

Alex-Waring committed:
ingress: Correct FromGroups rule Parsing

Currently fromGroups, which was added in https://github.com/cilium/cilium/pull/30708,
is correctly represented in the YAML but not being converted to a rule.
This commit fixes that by:
- Reporting that a rule requires derivation when an ingress component
  does, and making sure to clear ingress rules when creating that
  derivative.
- Adding FromGroups to parseToCiliumIngressCommonRule logic
- During validation make sure that fromGroups is not combined with other
  L3 rules.

Signed-off-by: Alex Waring <ajmwaring@gmail.com>

learnitall reviewed (approved):
The changes look good! I'm not too familiar with this part of cilium so it definitely would make sense to get more eyes on this.

Alex-Waring committed:
ingress: Testing for FromGroups

In the previous commit and https://github.com/cilium/cilium/pull/30708
the fromGroups resource was added, but testing was not covered properly.
This commit validates that we are working as expected.

Signed-off-by: Alex Waring <ajmwaring@gmail.com>
